osu.Game/OsuGame.cs

```diff
@@ -658,11 +659,14 @@ protected override void Dispose(bool isDisposing)
         }
 
         protected override IDictionary<FrameworkSetting, object> GetFrameworkConfigDefaults()
-            => new Dictionary<FrameworkSetting, object>
+        {
+            return new Dictionary<FrameworkSetting, object>
             {
-                // General expectation that osu! starts in fullscreen by default (also gives the most predictable performance)
-                { FrameworkSetting.WindowMode, WindowMode.Fullscreen }
+                // General expectation that osu! starts in fullscreen by default (also gives the most predictable performance).
+                // However, macOS is bound to have issues when using exclusive fullscreen as it takes full control away from OS, therefore borderless is default there.
+                { FrameworkSetting.WindowMode, RuntimeInfo.IsApple ? WindowMode.Borderless : WindowMode.Fullscreen }
```

Comment:
`IsApple` includes `iOS` – have you confirmed this doesn't break anything? If not, applying specifically to `RuntimeInfo.OS == Platform.macOS` may be better for safety.

Reply:
Fair enough, iOS runs on `Fullscreen` as its only supported window mode, so probably best to limit the conditional to macOS.